### ISSUE TITLE
lanes.cancel_error is not deep userdata

### ISSUE THREAD

**Issue by Skyb0rg007** [LuaLanes/lanes#138]
Simplest example:
```
#!/usr/bin/env luajit

local lanes = require'lanes'.configure()

local function foo()
    print(lanes.cancel_error)
end

lanes.gen("*", foo)()
```

``` luajit: /home/.../test.lua:9: can't copy non-deep full userdata across lanes ```

Because lanes.cancel_error is not deep userdata, it cannot be used to check the results of linda function calls within lanes

I'm sure there is a reason for this, however this is not documented and can cause confusion for people checking the output of these functions with ```lanes.cancel_error```

NOTE: I get the same error on Lua 5.3 (have not tested 5.1 or 5.2)

This is on Lanes version 3.10.1-1, the most recent version on luarocks

**Comment by sonoro1234** [comment]
I would say that is an very old issue "solved" this way
```
local lanes = require'lanes'.configure()

local function foo()
    local lanes = require"lanes"
    print(lanes.cancel_error)
end

lanes.gen("*", foo)()
```

**Comment by benoit-germain** [comment]
Well, this would warrant some investigation on my part to remove that particular limitation as `cancel_error` is a light userdata. There is no particular reason why it couldn't be copiable over across lanes. Only non-deep FULL userdata are protected against such attempts.

**Comment by benoit-germain** [comment]
Funnily enough it seems I can transfer cancel_error just fine with PUC-Lua...

**Comment by sonoro1234** [comment]
With Lua5.1 I get the same error with
```
local lanes = require'lanes'.configure()

local function foo()
    print(lanes.cancel_error)
end

lanes.gen("*", foo)()
```
```
C:\luaGL\sources\Lua5.1\lua.exe: C:\luaGL\sources\Lua5.1\lua\lanes.lua:332: can't copy non-deep full userdata
across lanes
stack traceback:
        [C]: in function 'core_lane_new'
        C:\luaGL\sources\Lua5.1\lua\lanes.lua:332: in function <C:\luaGL\sources\Lua5.1\lua\lanes.lua:330>
        C:\luaGL\gitsources\lanes\cancel_error.lua:8: in main chunk
        [C]: ?

```

**Comment by benoit-germain** [comment]
The problem is not that `cancel_error` isn't copyable in the above sample, it's that the module `lanes` is an upvalue of `foo`, which means that the lane creation mechanism attempts to transfer it, which tries to pull an internal lane handle created when `with_timers` is true (which is the default). Since lane handles are non-deep full userdata, you get this error.
One way for this to work is to change the code like so:
```lua
local lanes = require'lanes'.configure{with_timers = false}
```
Another is to avoid pulling the totality of the lanes module like so:
```lua
local lanes = require'lanes'.configure{with_timers = false}
local cancel_error = lanes.cancel_error

local function foo()
	print(cancel_error)
end

lanes.gen("*", foo)()
```
 So to sum up, this is not a bug, this is a feature :-).
